Mathlib/Data/Polynomial/Coeff.lean

```diff
@@ -293,16 +293,16 @@ theorem mul_X_pow_eq_zero {p : R[X]} {n : ℕ} (H : p * X ^ n = 0) : p = 0 :=
   ext fun k => (coeff_mul_X_pow p n k).symm.trans <| ext_iff.1 H (k + n)
 #align polynomial.mul_X_pow_eq_zero Polynomial.mul_X_pow_eq_zero
 
-theorem mul_X_pow_injective (n : ℕ) : Function.Injective fun P : R[X] => X ^ n * P := by
+-- TODO Unify this with `Polynomial.Monic.isRegular`
+theorem isRegular_X_pow (n : ℕ) : IsRegular ((X : R[X]) ^ n) := by
+  suffices : IsLeftRegular ((X : R[X]) ^ n)
+  · exact ⟨this, IsLeftRegular.right_of_commute (fun p ↦ commute_X_pow p n) this⟩
   intro P Q hPQ
   simp only at hPQ
   ext i
   rw [← coeff_X_pow_mul P n i, hPQ, coeff_X_pow_mul Q n i]
-#align polynomial.mul_X_pow_injective Polynomial.mul_X_pow_injective
 
-theorem mul_X_injective : Function.Injective fun P : R[X] => X * P :=
-  pow_one (X : R[X]) ▸ mul_X_pow_injective 1
-#align polynomial.mul_X_injective Polynomial.mul_X_injective
+@[simp] theorem isRegular_X : IsRegular (X : R[X]) := by simpa using isRegular_X_pow 1
```

Comment:
I guess you can probably prove these in the other direction and use `isRegular_X.pow n`?

Reply:
OK I've made this change but it doesn't really affect anything because of the existing API for polynomials.

Reply:
Update: looking at the diff now, I do prefer the new version (using `IsRegular.pow`).